campaignion_loqate/package/src/input.js

```diff
@@ -0,0 +1,76 @@
+/* globals: Drupal */
+
+/**
+ * Handler for a single input element and it’s wrapper.
+ */
+class Input {
+  constructor ($input, msg, dependent = []) {
+    this.$input = $input
+    this.$wrapper = $input.closest('.form-group')
+    this.dependent = dependent
+    this.formId = $input.closest('form').attr('id')
+    this.errorMessage = msg
+
+    this.valid = false
+    this.lastValue = ''
+    this.validated = false
+  }
+  bind () {
+    this.$input.on('change keyup', () => {
+      this.unmarkIfChanged()
+    })
+    return this
+  }
+  unmark () {
+    const validator = Drupal.myClientsideValidation.validators[this.formId]
+    const $wrapper = $('#clientsidevalidation-' + this.formId + '-errors')
+    const errors = validator.errorsFor(this.$input.get(0))
+    validator.addWrapper(errors).remove()
```

Comment:
Doesn’t this need something like `validator.hideErrors()`? :thinking:  
What does `.addWrapper().remove()` do?

Reply:
The code is mainly transferred from [here](https://github.com/moreonion/stripe_payment/blob/7.x-2.x/package/src/method-element.js#L123) and the [clientSideValidationErrorHandler](https://github.com/moreonion/stripe_payment/blob/7.x-2.x/package/src/method-element.js#L235) I don’t know if / when the code of [resetValidation](https://github.com/moreonion/stripe_payment/blob/7.x-2.x/package/src/method-element.js#L306).

Both are good questions, but I don’t know the stripe code close enough to answer them. Do you remember what these lines do and when they are needed?

Reply:
Have looked it up: if elements have a wrapper, `addWrapper(elements) adds the wrappers to elements so `hide()` will hide the errors *and* their wrappers. Sounds like the right choice here.

`hideErrors()` on the other hand will hide all errors on the form. I think our `resetValidation()` function was needed when switching between payment forms so you won’t get e.g. Stripe validation errors after switching to a DD form? :thinking: